internal/kubeclient/indexer.go

```diff
@@ -498,100 +496,48 @@ func indexStagesByWarehouse(obj client.Object) []string {
-		ServiceAccountsByOIDCGroupIndexField,
-		indexServiceAccountsByOIDCGroups,
+		ServiceAccountsByOIDCClaimIndexField,
+		indexServiceAccountsOIDCClaim,
 	)
 }
 
-// indexServiceAccountsByOIDCGroups is a client.IndexerFunc that indexes
-// ServiceAccounts by their OIDC group annotations.
-func indexServiceAccountsByOIDCGroups(obj client.Object) []string {
+// indexServiceAccountsOIDCClaim is a client.IndexerFunc that indexes ServiceAccounts by the
+// OIDC claim name and value annotations.
+func indexServiceAccountsOIDCClaim(obj client.Object) []string {
 	sa := obj.(*corev1.ServiceAccount) // nolint: forcetypeassert
-	rawGroups := strings.TrimSpace(sa.GetAnnotations()[rbacapi.AnnotationKeyOIDCGroups])
-	if rawGroups == "" {
-		return nil
-	}
-	groups := strings.Split(rawGroups, ",")
-	refinedGroups := make([]string, 0, len(groups))
-	for _, g := range groups {
-		if group := strings.TrimSpace(g); group != "" {
-			refinedGroups = append(refinedGroups, group)
+	rawClaimValues := map[string]string{}
+	for annotationKey, annotationValue := range sa.GetAnnotations() {
+		if strings.HasPrefix(annotationKey, rbacapi.AnnotationKeyOIDCClaimNamePrefix) {
+			rawClaimName := strings.ReplaceAll(annotationKey, rbacapi.AnnotationKeyOIDCClaimNamePrefix, "")
+			rawClaimValue := strings.TrimSpace(annotationValue)
+			if rawClaimValue == "" {
+				continue
+			}
+			rawClaimValues[rawClaimName] = rawClaimValue
 		}
 	}
-	return refinedGroups
-}
-
-// IndexServiceAccountsByOIDCSubjects sets up indexing of ServiceAccounts by
-// their OIDC subject annotations.
-//
-// It configures the cluster's field indexer to allow querying ServiceAccounts
-// using the ServiceAccountsByOIDCSubjectIndexField selector.
-func IndexServiceAccountsByOIDCSubjects(ctx context.Context, clstr cluster.Cluster) error {
-	return clstr.GetFieldIndexer().IndexField(
-		ctx,
-		&corev1.ServiceAccount{},
-		ServiceAccountsByOIDCSubjectIndexField,
-		indexServiceAccountsByOIDCSubjects,
-	)
-}
-
-// indexServiceAccountsByOIDCSubjects is a client.IndexerFunc that indexes
-// ServiceAccounts by their OIDC subject annotations.
-func indexServiceAccountsByOIDCSubjects(obj client.Object) []string {
-	sa := obj.(*corev1.ServiceAccount) // nolint: forcetypeassert
-	rawSubjects := strings.TrimSpace(sa.GetAnnotations()[rbacapi.AnnotationKeyOIDCSubjects])
-	if rawSubjects == "" {
+	if len(rawClaimValues) == 0 {
 		return nil
 	}
-	subjects := strings.Split(rawSubjects, ",")
-	refinedSubjects := make([]string, 0, len(subjects))
-	for _, s := range subjects {
-		if subject := strings.TrimSpace(s); subject != "" {
-			refinedSubjects = append(refinedSubjects, subject)
+	refinedClaimValues := []string{}
+	for rawClaimName, rawClaimValue := range rawClaimValues {
+		claimValues := strings.Split(rawClaimValue, ",")
+		for _, e := range claimValues {
+			if claimValue := strings.TrimSpace(e); claimValue != "" {
+				refinedClaimValues = append(refinedClaimValues, rawClaimName+"/"+claimValue)
```

Comment:
`rawClaimName+"/"+claimValue` feels a little bit brittle because using this index requires you to know this little bit of the indexer's implementation.

It may be best to create some sort of helper function to at least ensure consistent formatting between the indexer and spots where the index is used.

Reply:
I added a function called FormatClaim to the indexer.go, I'll leave this unresolved as I think this is what you mean but I'm not 100% sure if it makes sense to live somewhere else or if I've misunderstood.